deprecatedflags/deprecatedflags.go

```diff
@@ -29,19 +31,20 @@ import (
 // and instead using the option triggers a deprecation warning. The
 // return code can be safely ignored and is only provided to support
 // replacing functions like flag.Int in a global variable section.
-func Add(name string) bool {
-	flag.Var(deprecated{name: name}, name, "This option is deprecated.")
+func Add(ctx context.Context, name string) bool {
```

Comment:
I think in this case it is better to pass a `klog.Logger`. It makes the purpose of the parameter clearer and I cannot imagine any scenario where command line flag parsing will need timeouts.